csrc/python_frontend/fusion_definition.cpp

```diff
@@ -169,6 +169,36 @@ void FusionDefinition::print(std::ostream& os) const {
   os << std::endl;
 }
 
+std::string FusionDefinition::toGraphviz(
+    IrGraphGenerator::DetailLevel detail_level) const {
+  TORCH_CHECK(
+      id().has_value(), "Cannot create graphviz until definition is complete.");
+
+  auto scheds = fusionCache()->queryFusionSchedules(id().value());
+
+  Fusion* fusion = scheds->auto_gen_schedules->fusion();
+
+  if (detail_level != IrGraphGenerator::DetailLevel::ComputeOnly) {
+    // When schedule is shown, we need to actually process each segment of a
+    // segmented Fusion
+    auto kernel_runtime =
+        scheds->auto_gen_schedules->getMostRecentKernelRuntime();
+    // TODO: handle segmented fusions here
+    if (kernel_runtime->isSegmented()) {
+      std::cout << "Segmented" << std::endl;
+    } else {
+      std::cout << "Not segmented" << std::endl;
+    }
```

Comment:
I'm not entirely sure how to grab scheduled fusions for all segments in case the fusion is segmented. This is not important if `detail_level='ComputeOnly'`, but in other cases it may be nice to return a list of dot source strings for segmented fusions.